crates/aptos/src/node/analyze/fetch_metadata.rs

```diff
@@ -72,6 +72,25 @@ impl FetchMetadata {
         }
     }
 
+    async fn get_transactions(
+        client: &RestClient,
+        start: u64,
+        last: u64,
+    ) -> Result<Vec<Transaction>> {
+        let mut result = Vec::new();
+        let mut cursor = start;
+        while cursor < last {
+            let limit = std::cmp::min(1000, last - cursor);
+            let mut current = client
+                .get_transactions(Some(cursor), Some(limit as u16))
+                .await?
+                .into_inner();
+            result.append(&mut current);
+            cursor += limit
+        }
+        Ok(result)
+    }
```

Comment:
So this is just pinning the maximum page size to 1000, and parsing all the transactions at that rate?

Reply:
Yes. 

This is needed just for finding validator set for the epoch change. 

We go over all NewBlockEvents, and search within transactions of the last block in the epoch to find the validator set